docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Datasources_Agroal.adoc

```diff
@@ -0,0 +1,149 @@
+[[agroal]]
+= Agroal configuration
+
+The Agroal subsystem allows the definition of datasources. Declaring a new datasource consists of two separate steps: provide a JDBC driver and define a datasource that references the driver you installed.
+
+The Agroal subsystem is provided by the https://agroal.github.io/[Agroal] project. For a detailed description of the available configuration properties, please consult the project documentation.
+
+[[agroal-driver-installation]]
+== JDBC Driver Installation
+
+The JDBC driver is referenced through it's java class. Agroal requires an implementation of java.sql.Driver, javax.sql.DataSource or javax.sql.XADataSource for XA datasources.
+
+[NOTE]
+Agroal will try to load an java.sql.Driver from the specified module if the _class_ is not defined
+
+[TIP]
+Any installed _driver_ provides an operation called _class-info_ that lists all the properties available for that particular class, that can be set in the _connection-factory_.
+
+[source, xml]
+----
+<subsystem xmlns="urn:jboss:domain:agroal:1.0">
+    [...]
+    <drivers>
+        <driver name="h2" module="com.h2database.h2" class="org.h2.Driver"/>
+    </drivers>
+</subsystem>
```

Comment:
We should be using CLI examples instead of XML. Users are suggested to not edit the raw XML.